lib/tests/streamlit/data_mocks.py

```diff
@@ -362,6 +362,18 @@ def __str__(self):
     }
 )
 
+
+_offset_types = ["L", "S", "T", "H", "D", "W", "W-FRI", "Q", "Q-MAY"]
+
+PERIOD_TYPES_DF = pd.DataFrame(
```

Comment:
Oh, just saw that this is actually not working as expected. Arrow only supports different types/subtypes on a column level. If you mix different types in the same column (e.g. `period[H]` and `period[D]`), Arrow will throw an error if columns contain different types. It's still showing correctly since we have a fallback to string, but this is not using the frontend code.

Reply:
We probably need to separate those types on a column level.  We might not fit everything into the viewpoint of the table, but I think this is fine for now and we can figure out a solution to get more into the snapshot later on.

Reply:
Good catch! I updated the tests so that each column contains a different type of period.